lib/crawler/content_engine/extractor.rb

```diff
@@ -0,0 +1,73 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+# frozen_string_literal: true
+
+module Crawler
+  module ContentEngine
+    module Extractor
+      def self.extract(rulesets, crawl_result)
+        fields = {}
+
+        rulesets.each do |ruleset|
+          next unless match_url_filters?(ruleset, crawl_result)
+
+          fields.merge!(execute_rule(ruleset, crawl_result))
```

Comment:
Do you wanna log something if some field got extracted multiple times for same field_name by a mistake?